quimb/tensor/tensor_core.py

```diff
@@ -1356,6 +1356,70 @@ def fuse(self, fuse_map, inplace=False):
 
     fuse_ = functools.partialmethod(fuse, inplace=True)
 
+    def unfuse(self, unfuse_map, shape_map, inplace=False):
+        """Reshape single indices into groups of multiple.
+
+        Parameters
+        ----------
+        fuse_map : dict_like or sequence of tuples.
+            Mapping like: ``{existing_ind: sequence of new inds, ...}`` or an
+            ordered mapping like ``[(old_ind_1, new_inds_1), ...]`` in which
+            case the output tensor's new inds will be ordered. In both cases
+            the new indices are created at the old index's position of the 
+            tensor's shape.
+
+        shape_map : dict_like or sequence of tuples
+            Mapping like: ``{old_ind: new_ind_sizes, ...}`` or an
+            ordered mapping like ``[(old_ind_1, new_ind_sizes_1), ...]`` .
+
+        Returns
+        -------
+        Tensor
+            The transposed, reshaped and re-labeled tensor.
+        """
+        t = self if inplace else self.copy()
+
+        if isinstance(unfuse_map, dict):
+            old_inds, new_unfused_inds = zip(*unfuse_map.items())
+        else:
+            old_inds, new_unfused_inds = zip(*unfuse_map)
+
+        # for each set of fused dims, group into product, then add remaining
+        new_inds = [[i] for i in t.inds]
+        new_dims = [[i] for i in t.shape]
+        for ix in range(len(old_inds)):
+            ind_pos = t.inds.index(old_inds[ix])
+            new_inds[ind_pos] = new_unfused_inds[ix]
+            new_dims[ind_pos] = shape_map[old_inds[ix]]
+
+        # flatten new_inds, new_dims
+        new_inds = tuple(itertools.chain(*new_inds))
+        new_dims = tuple(itertools.chain(*new_dims))
+
+        if t.left_inds is not None:
+            new_left_inds = [[i] for i in t.left_inds]
+            for ix in range(len(old_inds)):
+                try:
+                    ind_pos = t.left_inds.index(old_inds[ix])
+                    new_left_inds[ind_pos] = new_unfused_inds[ix]
+                except(ValueError):
+                    pass
+
+            # flatten new_left_inds
+            new_left_inds = tuple(itertools.chain(*new_left_inds))
```

Comment:
Possibly a more succinct way to do this kind of thing is:
```python
new_left_inds = []
for ix in t.left_inds:
    try:
        new_left_inds.extend(unfuse_map[ix])
    except KeyError:
        new_left_inds.append(ix)        
```
but no need to change, just a note if you wanted to avoid going via `index` etc!